### DISCUSSION TITLE
Using plugins from different versions of jspsych

### DISCUSSION THREAD

**Discussion by iair-embon** [jspsych/jsPsych#2973] [Q&A] [answered]
Hello!

I am using an experiment using a plugin that was developed using jspsych version 5.0.3. In this experiment, I would also need to upload videos. I didn't find a plugin that does it in that old version, but in the latest version of jspsych (7.3.1). When I want to use both plugins loading both libraries at the same time, it doesn't load the page, and throws errors:

"Uncaught TypeError: jsPsych.plugins is undefined"
"Uncaught MigrationError: The global `jsPsych` variable is no longer available in jsPsych v7. Please follow the migration guide at https://www.jspsych.org/7.0/support/migration-v7/ to update your experiment."

Is there any way to use plugins from different versions of jspsych? 
Thanks for your help :)

**Comment by jodeleeuw** [comment, marked as answer]
Hey @iair-embon,

In general the major versions (5, 6, 7, etc.) are not backwards compatible. If you want to use the plugin from version 5 you'll have to update it to match the version 7 template. The migration guide in the error message describes the things that need to change in order to make that switch, and if you run into any specific problems feel free to open a thread here.

**Reply by iair-embon** [reply]
Thank you!
